scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Anchor.scala

```diff
@@ -0,0 +1,15 @@
+package scala.meta
+package semantic
+
+import org.scalameta.data._
+import scala.meta.io._
+import scala.meta.prettyprinters._
+
+// NOTE: `start` and `end` are String.substring-style,
+// i.e. `start` is inclusive and `end` is not.
+// Therefore Anchor.end can point to the last character plus one.
+@data class Anchor(path: AbsolutePath, start: Int, end: Int) {
```

Comment:
How is Anchor different from a `Position.Range`? One issue I have with `Anchor` is that it cannot represent `Input.LabeledString`, which I think would be nice to support in the semanticdb, as I mention in semanticdb.proto.